src/context/WishlistContext.jsx

```diff
@@ -0,0 +1,27 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { useAuth } from "./";
+import { wishlistHandlers } from "./utils/wishlistUtils";
+
+const WishlistContext = createContext(null);
+
+const useWishlist = () => useContext(WishlistContext);
+
+const WishlistProvider = ({ children }) => {
+  const { encodedToken } = useAuth();
+  const [wishlistProducts, setWishlistProducts] = useState([]);
+
+  useEffect(() => {
+    encodedToken &&
+      wishlistHandlers.getWishlist(encodedToken, setWishlistProducts);
+  }, [encodedToken]);
+
```

Comment:
Suggestion:
1. wishlistHandlers could be renamed to wishlistHandler
2. you could destructure getWishlist and other similar methods from wishlistHandlers for cleanliness and ease of implementation